src/main/java/org/opentripplanner/model/plan/legreference/ScheduledTransitLegReference.java

```diff
@@ -123,4 +143,60 @@ public ScheduledTransitLeg getLeg(TransitService transitService) {
 
     return leg;
   }
+
+  /**
+   * Return false if the stop id in the reference does not match the actual stop id in the trip
+   * pattern.
+   * Return true in the specific case where the stop ids differ, but belong to the same parent
+   * station.
+   *
+   */
+  private boolean matchReferencedStopInPattern(
+    TripPattern tripPattern,
+    int stopPosition,
+    FeedScopedId stopId,
```

Comment:
```suggestion
    @Nullable FeedScopedId stopId,
```

Reply:
it is indeed temporarily nullable for backward compatibility, not sure if this should be documented though, that could be misleading?

Reply:
Well, null is a valid value and will not, for example, lead to an NPE so I would say it should be there. I will not insist on it.